syncer/syncer.go

```diff
@@ -890,6 +916,22 @@ func (s *Syncer) sync(ctx context.Context, queueBucket string, db *Conn, jobChan
 	}
 }
 
+// redirectStreamer redirects binlog stream to given position
+func (s *Syncer) redirectStreamer(pos mysql.Position) (streamer.Streamer, error) {
+	var (
+		bs  streamer.Streamer
+		err error
+	)
+	log.Infof("reset global streamer to position: %v", pos)
```

Comment:
seems this function only used to reset `globalStreamer`, how about put variable `globalStreamer` into `Syncer`, and this function reset `s. globalStreamer`?